ceno_zkvm/src/instructions/riscv/dummy/dummy_circuit.rs

```diff
@@ -67,21 +67,23 @@ impl<E: ExtensionField> DummyConfig<E> {
         circuit_builder: &mut CircuitBuilder<E>,
         codes: InsnCodes,
     ) -> Result<Self, ZKVMError> {
-        let (with_rs1, with_rs2, with_rd) = match (codes.format, codes.kind) {
-            // ECALL reads its syscall_id, then do nothing.
-            (_, InsnKind::EANY) => (true, false, false),
+        // ECALL can do everything.
+        let is_ecall = matches!(codes.kind, InsnKind::EANY);
+        let (with_rs1, with_rs2, with_rd) = match codes.format {
+            _ if is_ecall => (true, true, true),
```

Comment:
I felt like we should let the caller to decide how many registers they want to access.

Reply:
For example, `ecall/commit` only needs to access X5 and X10 (set `with_rs1 = true, with_rs2 = true`). While complex ecall may use X5 / X10 / X11 by setting `with_rs1 = true, with_rs2 = true, with_rd = true`.

Reply:
I made it configurable so it is easy to write restricted ecall circuit. The default option is everything at once.